packages/components/src/components/hive-navigation/index.tsx

```diff
@@ -507,3 +526,33 @@ function HamburgerButton() {
     </button>
   );
 }
+
+const TopOfSiteMarker = function TopOfSiteMarker({
+  onChange,
+  className,
+}: {
+  onChange: (scrolled: boolean) => void;
+  className?: string;
+}) {
+  const markerRef = useRef<HTMLDivElement>(null);
+  const onChangeRef = useRef(onChange);
+  onChangeRef.current = onChange;
+
+  useEffect(() => {
+    if (markerRef.current && 'IntersectionObserver' in window) {
+      const observer = new IntersectionObserver(entries => {
+        onChangeRef.current(entries[0].boundingClientRect.y < -2);
+      });
+      observer.observe(markerRef.current);
+    }
+  }, []);
```

Comment:
maybe we can just use this? https://usehooks.com/useintersectionobserver

otherwise need a [disconnect call](https://stackoverflow.com/a/67826055)on the observer

Reply:
How do you feel about adding the dependency on usehooks for this?

The implementation there is subtly different (https://github.com/uidotdev/usehooks/blob/main/index.js#L512-L541), and this is probably the only time we need the observer here, but I don't mind adding it.

Reply:
yeah I guess we don't need to add a dependency we can just make change in ours